shell/common/shell.cc

```diff
@@ -992,11 +992,27 @@ void Shell::OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) {
   }
 }
 
+std::set<std::string> Shell::misbehaving_message_channels_ = {};
+
 // |PlatformView::Delegate|
 void Shell::OnPlatformViewDispatchPlatformMessage(
     std::unique_ptr<PlatformMessage> message) {
   FML_DCHECK(is_setup_);
-  FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
+  if (!task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread()) {
```

Comment:
The heart of the change.

Previously this was only a DCHECK, which helps engine developers but not plugin developers/users.